pkg/flow/flow.go

```diff
@@ -261,7 +264,7 @@ func (c *Flow) LoadFile(file *File, args map[string]any) error {
 		},
 	}
 
-	diags := c.loader.Apply(argumentScope, file.Components, file.ConfigBlocks, c.opts.OnExportsChange)
+	diags := c.loader.Apply(argumentScope, file.Components, file.ConfigBlocks, c.opts.OnExportsChange, c.opts.IsModule)
```

Comment:
This should probably be passed when the loader is constructed (since it never changes) so we don't just keep piling on more and more arguments here. 

Honestly, `c.opts.OnExportsChange` should also probably be moved too.